router/router.go

```diff
@@ -2049,14 +1998,30 @@ func (rt *HandleT) backendConfigSubscriber() {
 						destination := &source.Destinations[i]
 						if destination.DestinationDefinition.Name == rt.destName {
 							if _, ok := rt.destinationsMap[destination.ID]; !ok {
-								rt.destinationsMap[destination.ID] = &routerutils.BatchDestinationT{Destination: *destination, Sources: []backendconfig.SourceT{}}
+								rt.destinationsMap[destination.ID] = &routerutils.BatchDestinationT{
+									Destination: *destination,
+									Sources:     []backendconfig.SourceT{},
+								}
 							}
 							rt.destinationsMap[destination.ID].Sources = append(rt.destinationsMap[destination.ID].Sources, *source)
 
 							rt.destinationResponseHandler = New(destination.DestinationDefinition.ResponseRules)
 							if value, ok := destination.DestinationDefinition.Config["saveDestinationResponse"].(bool); ok {
 								rt.saveDestinationResponse = value
 							}
+
+							// Config key "throttlingCost" is expected to have the eventType as the first key and the call type
+							// as the second key (e.g. track, identify, etc...) or default to apply the cost to all call types:
+							// dDT["config"]["throttlingCost"] = `{"eventType":{"default":1,"track":2,"identify":3}}`
+							if value, ok := destination.DestinationDefinition.Config["throttlingCost"].(map[string]interface{}); ok {
+								if _, ok := rt.throttlingCosts[destination.ID]; !ok {
+									if rt.throttlingCosts == nil {
+										rt.throttlingCosts = make(map[string]types.EventTypeThrottlingCost)
+									}
+
+									rt.throttlingCosts[destination.ID] = types.NewEventTypeThrottlingCost(value)
```

Comment:
All destinations in router share the same `DestinationDefinition`, thus we don't need to keep separate throttling costs per destination id. A single throttling cost would be sufficient